src/Convex.jl

```diff
@@ -1,62 +1,115 @@
 export conv,
-    dotsort, entropy, exp, geomean, hinge_loss, huber, inner_product, invpos
-export log_perspective,
+    dotsort,
+    entropy,
+    exp,
+    geomean,
+    hinge_loss,
+    huber,
+    inner_product,
+    invpos,
+    log_perspective,
     logisticloss,
     logsumexp,
     matrixfrac,
     neg,
-    norm2,
     norm_1,
     norm_inf,
-    nuclearnorm
-export partialtrace,
-    partialtranspose, pos, qol_elementwise, quadform, quadoverlin, rationalnorm
-export relative_entropy,
-    sigmamax, square, sumlargest, sumlargesteigs, sumsmallest, sumsquares
-export GeomMeanHypoCone,
+    nuclearnorm,
+    partialtrace,
+    partialtranspose,
+    pos,
+    qol_elementwise,
+    quadform,
+    quadoverlin,
+    rationalnorm,
+    relative_entropy,
+    sigmamax,
+    square,
+    sumlargest,
+    sumlargesteigs,
+    sumsmallest,
+    sumsquares,
+    GeomMeanHypoCone,
     GeomMeanEpiCone,
     RelativeEntropyEpiCone,
     quantum_relative_entropy,
-    quantum_entropy
-export trace_logm, trace_mpower, lieb_ando
-
-export DCPViolationError
-
-# rexports from LinearAlgebra
-export diag, diagm, Diagonal, dot, eigmax, eigmin, kron, logdet, norm, tr
-
-# Constraints
-export Constraint
-export isposdef, ⪰, ⪯ # PSD constraints
-export socp
-export Constraint # useful for making abstractly-typed vectors via `Constraint[]`
-
-# Variables
-export constant, ComplexVariable, HermitianSemidefinite, Semidefinite, Variable
-export curvature,
-    evaluate, fix!, free!, monotonicity, sign, vexity, problem_vexity
-export BinVar, IntVar, ContVar, vartype, vartype!
-export get_constraints, add_constraint!, set_value!, evaluate
-
-# Signs
-export Positive, Negative, ComplexSign, NoSign
-
-# Problems
-export add_constraints!, maximize, minimize, Problem, satisfy, solve!
-export write_to_file
+    quantum_entropy,
+    trace_logm,
+    trace_mpower,
+    lieb_ando,
+    # Constraints
+    Constraint,
+    isposdef,
+    ⪰,
+    ⪯,
+    socp,
+    Constraint, # useful for making abstractly-typed vectors via `Constraint[]`
+    # Variables
+    constant,
+    ComplexVariable,
+    HermitianSemidefinite,
+    Semidefinite,
+    Variable,
+    curvature,
+    evaluate,
+    fix!,
+    free!,
+    monotonicity,
+    vexity,
+    problem_vexity,
+    BinVar,
+    IntVar,
+    ContVar,
+    vartype,
+    vartype!,
+    get_constraints,
+    add_constraint!,
+    set_value!,
+    evaluate,
+    # Signs
+    Positive,
+    Negative,
+    ComplexSign,
+    NoSign,
+    # Problems
+    add_constraints!,
+    maximize,
+    minimize,
+    Problem,
+    satisfy,
+    solve!,
+    write_to_file,
+    DCPViolationError
+
+# Imports and exports as needed to maintain backwards compatibility
+for k in (
+    :diag,
+    :diagm,
+    :Diagonal,
+    :dot,
+    :eigmax,
+    :eigmin,
+    :logdet,
+    :norm,
+    :norm2,
+    :tr,
+)
+    @eval begin
+        using LinearAlgebra: $k
+        export $k
+    end
+end
+using AbstractTrees: children
+using Base: sign
```

Comment:
Why is this line needed ? You explicitly do `Base.sign`.

Reply:
Because docs and tests assume that there is `Convex.sign`. Same for `Convex.children`. I didn't add a comment because this is tested.

Reply:
Ah ok, so we need it for backward compatibility

Reply:
Yeah. Before tagging the next release, there are a bunch of deprecation things we could remove. This would be one candidate.

Reply:
But that's for a different PR.